### PR TITLE
Improve api docs

### LINKED ISSUE #510: Document query placeholder syntax
Would have helped with #500

### LINKED ISSUE #522: Document boolean comparisons
From https://realm.io/docs/react-native/latest/#queries: 

"At the moment only a subset of the NSPredicate syntax is supported in the query language. Basic comparison operators ==, !=, >, >=, <, and <= are supported for numeric properties. ==, BEGINSWITH, ENDSWITH, and CONTAINS are supported for string properties. String comparisons can be made case insensitive by appending [c] to the operator: ==[c], BEGINSWITH[c] etc. Filtering by properties on linked or child objects can by done by specifying a keypath in the query eg car.color == 'blue'."

Please add something to mention booleans are compared using == etc., for example
- let purebredDogs = dogs.filtered('isPurebred == true');
  to avoid people (like me) trying, for example
- let purebredDogs = dogs.filtered('isPurebred');
- let purebredDogs = dogs.filtered('isPurebred == "true");

P.S. Realm has been working great for me :-)

### LINKED ISSUE #806: Guide does not mention how to do substitutions in queries
Can you use `%@` as in NSPredicate or `$x`? It is not mentioned anywhere in the guide, but the API reference says `$x` is used for placeholders.

We should update the guide, and check if the NSPredicate syntax is supported as well?

### PR DESCRIPTION
**Pull request by kristiandupont**
Resolves #510, resolves #806, resolves #522.

### CODE REVIEW

File: docs/tutorials/query-language.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+The Realm JavaScript SDK supports querying based on a language inspired by [NSPredicate](https://realm.io/news/nspredicate-cheatsheet/). 
+
+You query your Realm using the {@link Realm.Collection#filtered Collection.filtered()} method:
+
+```JS
+let dogs = realm.objects('Dog');
```
Comment (alazier) [inline docs/tutorials/query-language.md]: I really hate that we use dogs i all of our examples. I take partial responsibility for this. Maybe replace with Apps or Customers?

File: docs/tutorials/query-language.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+The Realm JavaScript SDK supports querying based on a language inspired by [NSPredicate](https://realm.io/news/nspredicate-cheatsheet/). 
+
+You query your Realm using the {@link Realm.Collection#filtered Collection.filtered()} method:
```
Comment (alazier) [inline docs/tutorials/query-language.md]: All the docs might sound better if we didn't use the word `You` and were instead passive:
`The {@link Realm.Collection#filtered Collection.filtered()} method is used to query a Realm`

File: docs/tutorials/query-language.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+```JS
+let dogs = realm.objects('Dog');
+let tanDogs = dogs.filtered('color == "tan" AND name BEGINSWITH "B"');
+```
+
+You can filter by linked or child objects with a keypath.
```
Comment (alazier) [inline docs/tutorials/query-language.md]: `You can` -> `It's possible to`

File: docs/tutorials/query-language.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+Example:
+```JS
+let johnsDogs = realm.Object('Dog').filtered('owner.name == "John"');
+```
+
+You can use numbered (`$0`, `$1`, ...) placeholders in your query string, using the succeeding parameters as the values.
```
Comment (alazier) [inline docs/tutorials/query-language.md]: remove `You can`, and `,` later in the sentence isn't necessary

File: docs/tutorials/query-language.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Note that for boolean properties, you should test against the expected keyword.
+
+Example:
+```JS
+let women = realm.objects('Person').filtered('isMale == false');
+```
+
+### String operators
+With strings, besides from the normal equality operators, you can use `BEGINSWITH`, `ENDSWITH` and `CONTAINS`.
```
Comment (alazier) [inline docs/tutorials/query-language.md]: Wording here is a bit odd. Change to
```
For string properties, prefix, suffix, and substring queries are supported by using the `BEGINSWITH`, `ENDSWITH`, and `CONTAINS`operators.
```

File: docs/tutorials/query-language.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+For any string operation, you can append `[c]` to the operator to make it case insensitive.
```
Comment (alazier) [inline docs/tutorials/query-language.md]: No comma needed

File: docs/tutorials/query-language.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+Example:
+```JS
+let peopleWhoseNameContainsA = realm.objects('Person').filtered('name CONTAINS[c] "a"');
+let Johns = realm.objects('Person').filtered('name ==[c] "john"');
+```
+
+### Composition
+You can use parentheses and the `&&`/`AND` and `||`/`OR` operators to compose queries. You can negate a predicate with `!`/`NOT`.
```
Comment (alazier) [inline docs/tutorials/query-language.md]: Remove `You can`

File: docs/tutorials/query-language.md (hunk 1 above)
Comment (kristiandupont) [inline docs/tutorials/query-language.md]: You are absolutely right, I hate that as well and I want to change it everywhere. I'll start here 😄

File: docs/conf.json [hunk 8]
```diff
@@ -14,9 +14,13 @@
   },
   "plugins": ["plugins/markdown"],
   "templates": {
+		"applicationName": "Realm JavaScript",
```
Comment (fealebenpae) [inline docs/conf.json]: Indentation

**Pull request merged**